modules/custom/az_finder/src/Form/AZFinderSettingsForm.php

```diff
@@ -0,0 +1,412 @@
+      if (!isset($override['origin'])) {
+        $override['origin'] = 'session';
+      }
+      $normalized_session_overrides[$key] = $override;
+    }
+
+    // Combine overrides.
+    $overrides = array_merge($config_overrides, $normalized_session_overrides);
+
+    $form['az_finder_tid_widget']['overrides'] = [
+      '#type' => 'container',
+      '#prefix' => '<div id="js-overrides-container">',
+      '#suffix' => '</div>',
+    ];
+
+    $form['az_finder_tid_widget']['overrides']['select_view_display_container'] = [
+      '#type' => 'container',
+      'select_view_display' => [
+        '#type' => 'select',
+        '#title' => $this->t('Add an Override'),
+        '#description' => $this->t('Select a particular filter widget to override the default display for each taxonomy term.'),
+        '#weight' => -1,
+        '#options' => $this->azFinderViewOptions->getViewOptions(),
+        '#empty_option' => $this->t('- Select -'),
+        '#attributes' => [
+          'id' => 'js-az-select-view-display',
+        ],
+      ],
+      'override' => [
+        '#type' => 'submit',
+        '#value' => $this->t('Override'),
+        '#ajax' => [
+          'callback' => '::ajaxAddOverride',
+          'wrapper' => 'js-overrides-container',
+          'effect' => 'fade',
+        ],
+        '#submit' => ['::submitOverride'],
+        '#attributes' => [
+          'class' => [
+            'button',
+            'button--primary',
+            'button--small',
+          ],
+        ],
+        '#states' => [
+          'disabled' => [
+            ':input[name="az_finder_tid_widget[overrides][select_view_display_container][select_view_display]"]' => ['value' => ''],
+          ],
+        ],
+      ],
+    ];
+
+    // Add override sections.
+    foreach ($overrides as $override) {
+      $this->addOverrideSection($form, $form_state, $override);
+    }
+
+    // Save combined overrides to the form state.
+    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * Separate submit handler for the override button.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function submitOverride(array &$form, FormStateInterface $form_state) {
+    // Retrieve selected view and display.
+    $selected_view_display = $form_state->getValue([
+      'az_finder_tid_widget',
+      'overrides',
+      'select_view_display_container',
+      'select_view_display',
+    ]);
+    [$view_id, $display_id] = explode(':', $selected_view_display);
+
+    // Prepare the configuration key.
+    $config_name = "az_finder.tid_widget.$view_id.$display_id";
+
+    // Initialize or load existing configuration.
+    $config = $this->configFactory->getEditable($config_name);
+
+    // Save the configuration.
+    $config->save();
```

Comment:
```suggestion
    $config->save();
    $form_state->setRebuild(TRUE);
```
I believe that `ajaxAddOverride()` is not a valid time to make modifications to the form. I think that the issue might be that at the time the dropdown option is chosen, that dropdown does not exist yet in the last built copy of the form, so might be silently ignored on submit. Rebuilding the form after submit seems to make saved override actually appear, because the override dropdowns have a chance to exist in a built copy of the form before the second submit happens.

If this correct (and doesn't introduce other bugs) the ajax callsbacks can probably be simplified.

Reply:
This seems to resolve the issue and @tadean will look into this further.

Reply:
Since `select_view_display_container` is also inside `overrides`, it seems perhaps like some filtering is needed to prevent this from being treated like an override.